tests/test_client.py

```diff
@@ -397,7 +397,7 @@ def test_path_is_not_double_encoded(self):
         self.assertEqual(req.path, "/get/test%20case")
 
         req = ClientRequest('get', "http://0.0.0.0/get/test%2fcase")
-        self.assertEqual(req.path, "/get/test%2fcase")
+        self.assertEqual(req.path, "/get/test/case")
```

Comment:
@TigorC %2f is indeed quoted `/`, but after unquoting you loose that information. A better example:
```
req = ClientRequest('get', "http://0.0.0.0/get/test%2fcase/passed")
```

What's req.path value should be? `test%2fcase/passed` or `test/case/passed`? Because you want to have a latter which is not what was requested.

RFC statement about what should be decoded and what is not is pretty clear on this.